src/main/java/gift/application/ProductService.java

```diff
@@ -0,0 +1,73 @@
+package gift.application;
+
+import gift.dao.ProductRepository;
+import gift.domain.Product;
+
+import gift.dto.ProductRequest;
+import gift.dto.ProductResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+
+    private final ProductRepository productRepository;
+
+    @Autowired
+    public ProductService(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    public List<ProductResponse> getAllProducts() {
+        List<Product> productList = productRepository.findAll();
+        List<ProductResponse> responseList = new ArrayList<>();
+        for (Product product : productList) {
+            responseList.add(new ProductResponse(product));
+        }
+        return responseList;
+    }
+
+    public ProductResponse getProductById(Long id) {
+        Product product = productRepository.findById(id).orElseThrow(
+                () -> new IllegalArgumentException("해당 상품은 존재하지 않습니다")
+        );
+        return new ProductResponse(product);
+    }
+
+    public ProductResponse createProduct(ProductRequest request) {
+        try {
+            Optional<Product> product = productRepository.findById(request.id());
+        } catch (Exception exception) {
+            return new ProductResponse(productRepository.save(new Product(request)));
+        }
```

Comment:
findById 에서 정상적으로 optional 반환이 된다면 코드를 이렇게도 변경할 수 있을 것 같습니다😀
1. 이미 존재하는지 데이터 체크
2. 존재하면 예외 반환 / 존재하지 않으면 저장
```suggestion
        productRepository.findById(request.id())
            .ifPresent(product -> {
                throw new IllegalArgumentException("해당 상품은 이미 존재하고 있습니다.");
            });

        return new ProductResponse(productRepository.save(new Product(request)));
```